app/knight/knight.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(
+            self,
+            knight_dict: dict
+    ) -> None:
+        self.name = knight_dict.get("name")
+        self.power = knight_dict.get("power")
+        self.hp = knight_dict.get("hp")
+        self.weapon = knight_dict.get("weapon")
+        self.armour = knight_dict.get("armour")
+        self.potion = knight_dict.get("potion")
+        self.battle_ready_stats = self.battle_ready_stats()
+
+    # calculates battle ready knight's stats
+    def battle_ready_stats(self) -> dict:

```

Comment:
Function names should be descriptive enough to not require additional comments describing their functionality. Try using verbs in function naming (e.g. `apply_potion`) and use docstrings when needed instead of comments.